base_report_to_printer/models/printing_server.py

```diff
@@ -98,41 +98,43 @@ def update_printers(self, domain=None, raise_on_error=False):
-            existing_printers = {
-                printer.system_name: printer for printer in server.printer_ids
-            }
             updated_printers = []
-            for name, printer_info in printers.items():
-                printer = self.env["printing.printer"]
-                if name in existing_printers:
-                    printer = existing_printers[name]
+            try:
+                connection = server._open_connection(raise_on_error=raise_on_error)
+                if not connection:
+                    server.printer_ids.write({"status": "server-error"})
+                    res = False
+                    continue
 
-                printer_values = printer._prepare_update_from_cups(
-                    connection, printer_info
-                )
-                if server != printer.server_id:
-                    printer_values["server_id"] = server.id
-
-                updated_printers.append(name)
-                # We want to keep any existing customized name over existing printer
-                # We want also to rely in the system name as a fallback to avoid
-                # empty names.
-                if not printer_values.get("name") and not printer.name:
-                    printer_values["name"] = name
-                if not printer:
-                    printer_values["system_name"] = name
-                    printer.create(printer_values)
-                elif printer_values:
-                    printer.write(printer_values)
+                # Update Printers
+                printers = connection.getPrinters()
+                existing_printers = {
+                    printer.system_name: printer for printer in server.printer_ids
+                }
+                for name, printer_info in printers.items():
+                    printer = self.env["printing.printer"]
+                    if name in existing_printers:
+                        printer = existing_printers[name]
 
+                    printer_values = printer._prepare_update_from_cups(
+                        connection, printer_info
+                    )
+                    if server != printer.server_id:
+                        printer_values["server_id"] = server.id
+
+                    updated_printers.append(name)
+                    # We want to keep any existing customized name over existing printer
+                    # We want also to rely in the system name as a fallback to avoid
+                    # empty names.
+                    if not printer_values.get("name") and not printer.name:
+                        printer_values["name"] = name
+                    if not printer:
+                        printer_values["system_name"] = name
+                        printer.create(printer_values)
+                    elif printer_values:
+                        printer.write(printer_values)
+            except Exception:
+                _logger.warning(_("Connection failed with %s", server.name))
```

Comment:
A little bit more context

```suggestion
                _logger.warning(_("Connection failed with printing server %s", server.name))
```